view_user_birthdate.py

```diff
@@ -0,0 +1,251 @@
+"""Build a dictionary of usernames and User objects. Additionally, take
+in user input, match it to any usernames in the dictionary and return
+that User's formatted birthdate based on the User's nationality.
+"""
+import re
+from get_input import get_input
+
+REGEX_DATE_RULE = ("(?P<month>[0-9]{1,2})" # Month.
+                   "/(?P<day>[0-9]{1,2})" # Day.
+                   "/(?P<year>[0-9]{4})" # Year.
+                  )
+
+class Date(object):
+    """Take in a string in the format:
+    mm/dd/yyyy
+    Parse it and create a Date object with the string attributes
+    'month', 'day', and 'year'. A Date object can return its attributes
+    using its display_date() method.
+    """
+    def __init__(self, string):
+        """Instantiate a Date object with the attributes 'month',
+        'day', and 'year'.
+        """
+        assert_response = "A string in the format mm/dd/yyyy must be provided."
+        assert re.search(REGEX_DATE_RULE, string), assert_response
+        self._month, self._day, self._year = parse_date(string)
+
+    def get_date_attributes(self):
```

Comment:
for the template method pattern i mentioned elsewhere to prevent this getter, you'd write something like

```
def display_date(self):
    return self._format_date(self._month, self._day, self._year)

@abstractmethod
def _format_date(self, month, day, year):
    raise NotImplementedError()
```
you'd write the above in Date.

In American, European and ChineseDate, you'd write
```
def _format_date(self, month, day, year):
   ...do the formatting logic here...
```

In this way, the parent class never has its private variables 'gotten'. It instead 'tells' its children by calling them and *providing* its private variables. This has at least two benefits:
1. Children don't know how the parent is implemented (reduced coupling)
2. Children are easier to test since a test can just pass dummy values in

This pattern is also close to the 'strategy pattern' you might be interested in reading up on.